openedx/features/wikimedia_features/messenger/api/v0/views.py

```diff
@@ -105,7 +106,17 @@ class InboxView(viewsets.ModelViewSet):
     search_fields = ['last_message__sender__username', 'last_message__receiver__username']
 
     def get_queryset(self):
-        return Inbox.user_inbox.find_all(self.request.user)
+        qs = Inbox.user_inbox.find_all(self.request.user)
+        # sort qs to show inbox with unread messages on top
+        return sorted(
+            qs,
+            key=lambda x, user=self.request.user: x.unread_count if (x.unread_count and x.last_message.receiver==user) else 0,
+            reverse=True
+        )
+
+    def get_object(self, *args, **kwargs):
```

Comment:
have to override this as sorted(..) will return list of objects instead of querryset but get_object expect queryset.